pkg/controllers/machine/disruption/drift.go

```diff
@@ -82,13 +87,24 @@ func (d *Drift) Reconcile(ctx context.Context, provisioner *v1alpha5.Provisioner
 
 // isDrifted will check if a machine is drifted from the fields in the provisioner.Spec and
 // the cloudprovider
-func (d *Drift) isDrifted(ctx context.Context, provisioner *v1alpha5.Provisioner, machine *v1alpha5.Machine) (bool, error) {
+func (d *Drift) isDrifted(ctx context.Context, provisioner *v1alpha5.Provisioner, machine *v1alpha5.Machine) (string, error) {
+	var reason cloudprovider.DriftReason
 	cloudProviderDrifted, err := d.cloudProvider.IsMachineDrifted(ctx, machine)
 	if err != nil {
-		return false, err
+		return "", err
+	}
+
+	if cloudProviderDrifted != "" {
```

Comment:
Do we need to check each of these individually if one of them is true? Can we just short-circuit and `return`?